src/vs/workbench/parts/extensions/node/extensionsWorkbenchService.ts

```diff
@@ -607,10 +612,15 @@ export class ExtensionsWorkbenchService implements IExtensionsWorkbenchService,
 	}
 
 	private isAutoUpdateEnabled(): boolean {
-		return this.configurationService.getValue(AutoUpdateConfigurationKey);
+		const configValue = this.configurationService.getValue(AutoUpdateConfigurationKey);
+		return configValue === true || configValue === 'checkAndInstall';
 	}
 
 	private eventuallySyncWithGallery(immediate = false): void {
+		if (this.configurationService.getValue(AutoUpdateConfigurationKey) === 'off') {
```

Comment:
Name of this method is confusing with the new setting. This should be called `checkForUpdatesAutomatically`